src/providers/WalletProvider.tsx

```diff
@@ -0,0 +1,94 @@
+
+interface WalletContextType {
+  address?: string;
+  network?: string;
+  networkPassphrase?: string;
+};
+
+const WalletContext = createContext<WalletContextType | undefined>(undefined);
+
+export const WalletProvider = ({ children }: { children: React.ReactNode }) => {
+  const [address, setAddress] = useState<string>();
+  const [network, setNetwork] = useState<string>();
+  const [networkPassphrase, setNetworkPassphrase] = useState<string>();
+  
+  const nullify = () => {
+    setNetwork(undefined)
+    setNetworkPassphrase(undefined)
+    setAddress(undefined)
+  }
+
+  const updateCurrentWalletState = async () => {
+    // There is no way, with StellarWalletsKit, to check if the wallet is
+    // installed/connected/authorized. We need to manage that on our side by
+    // checking our storage item.
+    if (!storage.getItem("walletId")) {
+      nullify()
+    } else {
+
+      // If our storage item is there, then we try to get the user's address &
+      // network from their wallet. Note: `getAddress` MAY open their wallet
+      // extension, depending on which wallet they select!
+      try {
+        const [a, n] = await Promise.all([
+          wallet.getAddress(),
+          wallet.getNetwork(),
+        ])
+        if (a.address !== address) setAddress(a.address)
+        if (n.network !== network) setNetwork(n.network)
+        if (n.networkPassphrase !== networkPassphrase)
+          setNetworkPassphrase(n.networkPassphrase)
+      } catch (e) {
+        // If `getNetwork` or `getAddress` throw errors... sign the user out???
+        nullify()
+
+        // then log the error (instead of throwing) so we have visibility
+        // into the error while working on Scaffold Stellar but we do not
+        // crash the app process
+        console.error(e)
+      }
+    }
+  }
+
+  useEffect(() => {
+    let timer: NodeJS.Timeout
+    void (async () => {
+      // Poll the wallet extension for updates every second. This allows the
+      // app to stay aware of which address & network the user selected in
+      // their wallet extension.
+      //
+      // Using `while (true)` (as opposed to `setInterval`) allows awaits to
+      // complete between iterations.
+      while (true) {
+        await new Promise((res) => {
+          timer = setTimeout(res, 1000)
+        });
+        await updateCurrentWalletState()
+      }
+    })()
+
+    // Clear the timeout when the component unmounts
+    // (unmounting removes the `while (true)` loop)
+    return () => {
+      if (timer) clearTimeout(timer)
+    }
+  }, []) // eslint-disable-line react-hooks/exhaustive-deps -- it SHOULD only run once per component mount
+
+  const contextValue = useMemo(() => ({
+    address,
+    network,
+    networkPassphrase,
+  }), [address, network, networkPassphrase]);
+
+  return (
+    <WalletContext value={contextValue}>
+      {children}
+    </WalletContext>
+  );
+};
+
+export { WalletContext }
+export type { WalletContextType }
```

Comment:
These are exported at the end to get around the (erroneous?) linting issue

<img width="1164" alt="image" src="https://github.com/user-attachments/assets/322455f0-a01f-4485-bab4-14be3bd427a9" />

it's SO silly. I really want to be able to put the `export` inline. It makes no sense that moving it to a separate line appeases the linter. The behavior is the same both ways.

Maybe we should add an `eslint-disable-line` comment instead?

```ts
export const WalletContext = // eslint-disable-line react-refresh/only-export-components
  createContext<WalletContextType | undefined>(undefined);
```

(Already wondering how useful this linter setting is, if we can just ignore it in the only place it would be useful so far and everything works the same way 🤔 )

Reply:
I changed the underlying code to something close to my suggestion there:

```ts
export const WalletContext = // eslint-disable-line react-refresh/only-export-components
  createContext<WalletContextType>({});
```

It's at least somewhat more straightforward without the `| undefined` stuff. Not sure if there's a strong reason you're preferring to initialize the provider with `undefined` in #25, @pselle. Initializing with a (typed) empty object seems fine to me?